Carbon.Redis/Abstractions/ICarbonCacheExtensions.cs

```diff
@@ -136,28 +312,86 @@ public static async Task HashSet<T>(this ICarbonCache instance, string key, Dict
             return keyValuePairs;
         }
 
-        public static async Task SetAdd<T>(this ICarbonCache instance, string key, T value)
+
+        /// <summary>
+        /// Gets your entire information in a dictionary for all fields within a single redis hash key, if key does not exist, it tries to fill with the given function
+        /// </summary>
+        /// <typeparam name="T">The value type that you want to retrieve from given field/dictionary key, use object if all the types may differ, then convert it later</typeparam>
+        /// <param name="instance"></param>
+        /// <param name="key">Redis Key</param>
+        /// <param name="setMethodIfNotExists">Execute if this key does not exist</param>
+        /// <returns>Dictionary of the all hash fields</returns>
+        public static async Task<Dictionary<string, T>> HashGet<T>(this ICarbonCache instance, string key, Func<Task<Dictionary<string, T>>> setMethodIfNotExists) where T : class
         {
-            await instance.GetDatabase().SetAddAsync(key, value.ToByteArray(SerializationType));
+            var values = await instance.HashGet<T>(key);
```

Comment:
Is there a reason why we are not using PreferReplica flag in this get function? How should we decide if we should use that flag or not?

Reply:
It's another function that is handled in that static class library in-which handled by PreferReplica as well.